spec/features/paginate_posts_index_spec.rb

```diff
@@ -0,0 +1,14 @@
+require 'spec_helper'
+
+describe 'Paginate posts index' do
+  subject { page }
+  describe 'index is paginated' do
+    before do
+      25.times { FactoryGirl.create(:blog_post) }
+      visit blog_posts_path
+    end
+      BlogPost.page(1).each do |post|
```

Comment:
You don't really want to touch the inner workings of the app. You want to drive the user's browser in feature tests.

So you write feature tests by creating some context and then guiding the user  through the website and making assertions.

For example:

```
# Create more posts than what a page can hold
create_list(:blog_post, BlogPost::POSTS_PER_PAGE + 1)

# Drive the user
visit root_path
click_on "Blog"
expect(page).to have_selector('.blog-post', count: BlogPost::POSTS_PER_PAGE)
```